### PR TITLE
Clear web terminal when session ends

### LINKED ISSUE #6107: Clear web terminal after session idle timeout.
Right now, session idle timeout will terminate the session but not clear the web terminal. This has been requested for FedRAMP compliance.

### PR DESCRIPTION
**Pull request by atburke**
Only applies when Teleport is compiled for FedRAMP compliance. Closes #6107.

### CODE REVIEW

File: lib/client/terminal/terminal_unix.go [hunk 1]
```diff
@@ -152,6 +152,15 @@ func (t *Terminal) Stderr() io.Writer {
 	return t.stderr
 }
 
+// Clear clears the terminal, including scrollback.
+func (t *Terminal) Clear() error {
+	const resetPattern = "\x1b[3J\x1b\x63\n"
```
Comment (zmb3) [inline lib/client/terminal/terminal_unix.go:157]: Maybe leave a comment linking to the source of this escape code so we know where to look if we need to change it?

File: lib/client/terminal/terminal_windows.go [hunk 2]
```diff
@@ -272,6 +272,10 @@ func (t *Terminal) Stderr() io.Writer {
 	return t.stderr
 }
 
+func (t *Terminal) Clear() error {
+	return nil
```
Comment (zmb3) [inline lib/client/terminal/terminal_windows.go:276]: Does this need to work on Windows too?

**Review by zmb3 — APPROVED** [review]

File: lib/client/terminal/terminal_windows.go (hunk 2 above)
Comment (atburke) [reply lib/client/terminal/terminal_windows.go:276]: #6107 only asks for it to work in the web terminal.

File: lib/client/terminal/terminal_unix.go (hunk 1 above)
Comment (atburke) [reply lib/client/terminal/terminal_unix.go:157]: Good plan. @gzdunek can you add this?

File: lib/client/terminal/terminal_windows.go (hunk 2 above)
Comment (nklaassen) [reply lib/client/terminal/terminal_windows.go:276]: I actually don't know the answer to this, but would this windows terminal not show up in the web UI as well? Is there any reason we shouldn't make it work on Windows?

File: lib/client/terminal/terminal_unix.go (hunk 1 above)
Comment (russjones) [reply lib/client/terminal/terminal_unix.go:157]: I think some of it came from here: https://superuser.com/questions/122911/what-commands-can-i-use-to-reset-and-clear-my-terminal

But the above looks different, @gzdunek how did you arrive at that escape sequence?

File: lib/client/terminal/terminal_windows.go (hunk 2 above)
Comment (russjones) [reply lib/client/terminal/terminal_windows.go:276]: We should support all terminals. The Web UI was requested in the ticket, but after talking with the customer, it's a FedRAMP requirement for all terminals.

If the escape code approach works for the Web UI (as in xterm.js understands it) we should just make the changes in `lib/client` and let all terminals automatically pick it up.

File: lib/client/terminal/terminal_unix.go (hunk 1 above)
Comment (gzdunek) [reply lib/client/terminal/terminal_unix.go:157]: `/x1b[3J` - clears scrollback (https://newbedev.com/how-do-i-reset-the-scrollback-in-the-terminal-via-a-shell-command https://apple.stackexchange.com/a/187773) - it is needed at least for the Mac terminal
> If n is 3, clear entire screen and delete all lines saved in the scrollback buffer (this feature was added for xterm and is supported by other terminal applications)."


`\x1b\x63` - clears current screen - same as `'\0033\0143'` linked above

Add it as a comment?

File: lib/client/terminal/terminal_windows.go (hunk 2 above)
Comment (alex-kovoy) [reply lib/client/terminal/terminal_windows.go:276]: xtermjs does understand it

File: lib/client/terminal/terminal_unix.go (hunk 1 above)
Comment (gzdunek) [reply lib/client/terminal/terminal_unix.go:160]: Added

File: lib/client/terminal/terminal_windows.go (hunk 2 above)
Comment (atburke) [reply lib/client/terminal/terminal_windows.go:276]: I've tested the escape code in Windows. Command line and Powershell did not work (although [wikipedia](https://en.wikipedia.org/wiki/ANSI_escape_code#DOS,_OS/2,_and_Windows) suggests that they could with some fiddling), but Windows Terminal does.

Comment (russjones) [reply lib/client/terminal/terminal_windows.go:276]: @timothyb89 What does `tsh.exe` support? Also, any hints on how @atburke could handle this for Windows?

Comment (timothyb89) [reply lib/client/terminal/terminal_windows.go:276]: tsh does enable ANSI processing on Windows, so assuming `Terminal.Close()` hasn't yet been called (which appears to be the case) I'd expect the usual escape sequence to just work. Microsoft recommends the same: https://docs.microsoft.com/en-us/windows/console/clearing-the-screen

There are other fallback methods involving old Win32 APIs if we _really_ have to, but hopefully we can get the ANSI sequence working.

(Slight possible caveat that the ANSI sequences might not work on very early versions of Windows 10, as in v1609 if not earlier. Hopefully we don't need to worry about them w.r.t compliance issues?)

Comment (russjones) [reply lib/client/terminal/terminal_windows.go:276]: I agree, we should not fall back to Win32 or support very early version of Windows 10.

@atburke What version of Windows were you using?

Comment (atburke) [reply lib/client/terminal/terminal_windows.go:276]: This was on Windows 10 version 1809.

Comment (timothyb89) [reply lib/client/terminal/terminal_windows.go:276]: I just copied the impl over from `terminal_unix.go` to `terminal_windows.go` and it cleared my cmd.exe just fine (on Win10 20H2)

**Review by russjones — APPROVED** [review]

**Comment by atburke** [comment]
@timothyb89 Would you mind giving this a review when you have a moment?

**Review by timothyb89 — APPROVED** [review]
Ah, sorry about that, looks good to me!

**Pull request merged**